rust/tokens_transfer/src/lib.rs

```diff
@@ -0,0 +1,103 @@
+use std::cell::RefCell;
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+use std::ops::Deref;
+use candid::{candid_method, CandidType};
+
+use ic_cdk::api::call::call;
+use ic_cdk_macros::*;
+use ic_ledger_types::{AccountIdentifier, BlockIndex, DEFAULT_SUBACCOUNT, MAINNET_LEDGER_CANISTER_ID, Memo, Subaccount, Tokens};
+use ic_types::Principal;
+use serde::{Deserialize, Serialize};
+
+#[cfg(test)] use std::path::PathBuf;
+#[cfg(test)] use std::io::Write;
+use ic_cdk::id;
+
+use crate::errors::TransferError;
+
+#[cfg(test)] mod candid_util;
+mod errors;
+
+#[derive(CandidType, Serialize, Deserialize, Clone, Debug, Hash, PartialEq)]
+pub struct Conf {
+    ledger_canister_id: Principal,
+    subaccount: Option<Subaccount>,
+    transaction_fee: Tokens
+}
+
+impl Default for Conf {
+    fn default() -> Self {
+        Conf {
+            ledger_canister_id: MAINNET_LEDGER_CANISTER_ID,
+            subaccount: None,
+            transaction_fee: Tokens::from_e8s(10_000),
+        }
+    }
+}
+
+thread_local! {
+    static CONF: RefCell<Conf> = RefCell::new(Conf::default());
+}
+
+#[init]
+#[candid_method(init)]
+fn init(conf: Conf) {
+    CONF.with(|c| c.replace(conf));
+}
+
+#[derive(CandidType, Serialize, Deserialize, Clone, Debug, Hash)]
+pub struct TransferArgs {
+    amount: Tokens,
+    to_account: Principal,
```

Comment:
```suggestion
    to_principal: Principal,
```
I find it confusing: we identify accounts with 32-byte arrays, not principals.

Reply:
The idea is to also show how to create account id from principals.

Reply:
I applied the change with the last commit